src/components/ec-full-screen-overlay/ec-full-screen-overlay.vue

```diff
@@ -0,0 +1,105 @@
+<template>
+  <transition name="ec-full-screen-overlay__fade">
+    <div
+      class="ec-full-screen-overlay"
+      data-test="ec-full-screen-overlay"
+    >
+      <div
+        ref="overlayContainer"
+        class="tw-col-full"
+        data-test="ec-full-screen-overlay__container"
+      >
+        <div class="tw-grid tw-grid-cols-3">
+          <div class="tw-col-full sm:tw-col-3" />
+          <div class="tw-col-full sm:tw-col-6">
+            <div class="tw-grid">
+              <div class="tw-col-full sm:tw-col-9">
```

Comment:
We can just use a basic flex with 2 columns instead of grid, the one column for title can have flex-grow set to 1, the one for the icon to 0. Then we're sure that the icon only removes space as small as possible from the space needed for title and you you can also remove `tw-text-right` from the icon. Another thing is that having tw-col-9 only `min-width: 580px` breaks the title on small screens:

![Screenshot from 2022-11-03 18-07-51](https://user-images.githubusercontent.com/5777410/199800978-28d59f82-d852-4a48-96d7-acc628948ac7.png)